src/server.h

```diff
@@ -800,9 +800,11 @@ struct serverObject {
                               * and most significant 16 bits access time). */
     unsigned hasexpire : 1;
     unsigned hasembkey : 1;
+    unsigned hasembval : 1;
     unsigned refcount : OBJ_REFCOUNT_BITS;
-    void *ptr;
+    void *val_ptr;
 };
+static_assert(sizeof(struct serverObject) <= 16, "unexpected size - verify struct is packed correctly");
```

Comment:
To make this assert correct on 64-bit and 32-bit, how about computing it as `8 + sizeof(void *)`?

```suggestion
static_assert(sizeof(struct serverObject) <= 8 + size(void *), "unexpected size - verify struct is packed correctly");
```